stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -454,7 +454,10 @@ public struct Set<Element : Hashable> :
     return _variantStorage.endIndex
   }
 
-  // TODO: swift-3-indexing-model - add docs
+  /// Returns the index after the given one. 
+  ///
+  /// If this is the last valid index, returns endIndex.
+  /// Aborts on endIndex.
```

Comment:
That should just be `- Precondition: i != endIndex`.  We don't normally document the response to precondition violations; we want the flexibility to change it later.